app/controllers/games_controller.rb

```diff
@@ -25,12 +30,22 @@ def authorize
   end
 
   def set_game
-    @game = Game.joins(:srdc).find_by(speedrun_dot_com_games: {shortname: params[:game]}) || Game.find_by(id: params[:game])
+    @game = Game.joins(:srdc).find_by(speedrun_dot_com_games: {shortname: params[:game]}) ||
+            Game.find_by(id: params[:game])
+  end
 
+  def redirect_if_not_found
     if @game.nil?
       redirect_to games_path(q: params[:game])
       return
     end
     redirect_to game_path(@game) if @game.srdc.try(:shortname).present? && params[:game] == @game.id.to_s
   end
+
+  def set_games
+    @games = Hash.new([])
+    SpeedrunDotComGame.order('ASCII(name) ASC').each do |game|
+      @games[game.name[0].downcase] += [game]
```

Comment:
Purely out of curiosity, why do you append an array with the game as its only element?

Reply:
`+=` isn't technically an append, it creates a new array with the two values concatenated. Normally I'd append with `<<`, but using `+=` avoids the issue I talked with you about earlier:
```ruby
> h = Hash.new
=> {}
> h.default = []
=> []
> h['a'] << 'alice'
=> ["alice"]
> h
=> {}
> h.default
=> ["alice"]
> h['a']
=> ["alice"]
```
Using `+=` is more expensive because it copies the array, but avoids the above issue. It also requires the right-hand side be another array:
```ruby
> h['b'] += 'bob'
TypeError: no implicit conversion of String into Array
> h['b'] += ['bob']
=> ["alice", "bob"]
> h['b']
=> ["alice", "bob"]
```
However in writing this I realized we're doing this array copy for every game, so I'm changing this to be less expensive at the cost of an extra check.

Reply:
Ah, that makes sense.  I poked around and found you can set the default proc of the hash to return an array which would allow you to do the first way.
```ruby
h = Hash.new { |hash, key| hash[key] = [] }
h['a'] << 'alice'
```

Not sure if that is cleaner than just doing the check as you are now.